server/src/main/java/org/elasticsearch/index/engine/NoOpEngine.java

```diff
@@ -44,8 +44,7 @@ public NoOpEngine(EngineConfig config) {
     protected DirectoryReader open(final IndexCommit commit) throws IOException {
         final Directory directory = commit.getDirectory();
         final List<IndexCommit> indexCommits = DirectoryReader.listCommits(directory);
-        assert indexCommits.size() == 1 : "expected only one commit point";
```

Comment:
did this fail? for which reason?

Reply:
ping?

Reply:
Woops sorry, this assertion tripped in some CCR tests (I don't remember which ones exactly). This assertion has been introduced after https://github.com/elastic/elasticsearch/pull/33903/files#r219476623 but I don't think it is valid again and that having a single index commit for NoOpEngine is a requirement.

Reply:
While having a single commit is not strictly necessary for NoOpEngine, it would be desirable if closing an index would always get us to that point. I think we should investigate this in a follow-up. Can you add a bullet to the meta issue?

Reply:
Sure, I added a point in https://github.com/elastic/elasticsearch/issues/33888